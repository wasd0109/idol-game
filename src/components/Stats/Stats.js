import React from 'react';

import profileImage from '../../assets/profile_image.jpg';
import Skeleton from 'react-loading-skeleton';

function Stats({ stats }) {
  return (
    <div className="md:w-4/12" id="stats">
      <ul className="flex flex-wrap border-l-2 border-t-2">
        <div className="w-12/12 h-auto border-l-0 border-r-2 border-b-2 m-auto">
          <img src={profileImage} alt="Profile" />
        </div>
        {stats.map((stat, i) => {
          if (stat[0] === 'id' || stat[0] === 'userid' || stat[0] === 'message')
            return null;
          return stat[1] ? (
            <li
              key={stat[0]}
              className="w-6/12 capitalize border-r-2 border-b-2"
            >
              {stat[0]} <span className="text-lg font-bold">{stat[1]}</span>
            </li>
          ) : (
            <li key={i} className="w-6/12 capitalize border-r-2 border-b-2">
              {stat[0]}{' '}
              <span className="text-lg font-bold">
                <Skeleton width={40} />
              </span>
            </li>
          );
        })}
      </ul>
    </div>
  );
}

export default Stats;
