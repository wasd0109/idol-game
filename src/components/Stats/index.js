import Profile from './Stats';
export default Profile;
