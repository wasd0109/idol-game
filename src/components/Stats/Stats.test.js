import React from 'react';
import { render, screen } from '@testing-library/react';
import Stats from './Stats';

const initialProps = {
    stats: [["name", ""], ["title", ""], ["HP", 0]],
}


let component;


describe("Stats component render correctly", () => {
    beforeEach(() => {
        component = render(<Stats {...initialProps} />)
    })
    test("Stats component render", () => {
        expect(component).toMatchSnapshot();
    })

    test.each(initialProps.stats)("Stats display skeleton before player info fetched", (expectedStat) => {

        expect(screen.getByText(String(expectedStat)).children[0].id).toEqual("skeleton")
    })
})

describe("Stats component function correctly", () => {
    const props = {
        stats: [["name", "Ken"], ["title", "Rookie"], ["HP", 25]],
    }
    test.each(props.stats)("Stats display correctly", (statName, stat) => {
        render(<Stats {...props} />)
        expect(screen.getByText(String(statName))).toBeInTheDocument();
        expect(screen.getByText(String(stat))).toBeInTheDocument();
    })
})

