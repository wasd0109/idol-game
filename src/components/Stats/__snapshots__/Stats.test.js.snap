// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats component render correctly Stats component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="flex flex-wrap border-l-2 border-t-2"
      >
        <div
          class="w-12/12 h-auto border-l-0 border-r-2 border-b-2 m-auto"
        >
          <img
            alt="Profile"
            src="profile_image.jpg"
          />
        </div>
        <li
          class="w-6/12 capitalize border-r-2 border-b-2"
        >
          name
           
          <span
            class="text-lg font-bold"
            id="skeleton"
          >
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                style="width: 40px;"
              >
                ‌
              </span>
            </span>
          </span>
        </li>
        <li
          class="w-6/12 capitalize border-r-2 border-b-2"
        >
          title
           
          <span
            class="text-lg font-bold"
            id="skeleton"
          >
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                style="width: 40px;"
              >
                ‌
              </span>
            </span>
          </span>
        </li>
        <li
          class="w-6/12 capitalize border-r-2 border-b-2"
        >
          HP
           
          <span
            class="text-lg font-bold"
            id="skeleton"
          >
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                style="width: 40px;"
              >
                ‌
              </span>
            </span>
          </span>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="flex flex-wrap border-l-2 border-t-2"
    >
      <div
        class="w-12/12 h-auto border-l-0 border-r-2 border-b-2 m-auto"
      >
        <img
          alt="Profile"
          src="profile_image.jpg"
        />
      </div>
      <li
        class="w-6/12 capitalize border-r-2 border-b-2"
      >
        name
         
        <span
          class="text-lg font-bold"
          id="skeleton"
        >
          <span>
            <span
              class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              style="width: 40px;"
            >
              ‌
            </span>
          </span>
        </span>
      </li>
      <li
        class="w-6/12 capitalize border-r-2 border-b-2"
      >
        title
         
        <span
          class="text-lg font-bold"
          id="skeleton"
        >
          <span>
            <span
              class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              style="width: 40px;"
            >
              ‌
            </span>
          </span>
        </span>
      </li>
      <li
        class="w-6/12 capitalize border-r-2 border-b-2"
      >
        HP
         
        <span
          class="text-lg font-bold"
          id="skeleton"
        >
          <span>
            <span
              class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              style="width: 40px;"
            >
              ‌
            </span>
          </span>
        </span>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
